Source/JavaScriptCore/wasm/WasmSectionParser.cpp

```diff
@@ -730,8 +760,11 @@ auto SectionParser::parseStructType(uint32_t position, RefPtr<TypeDefinition>& s
     WASM_PARSER_FAIL_IF(!fields.tryReserveCapacity(fieldCount), "can't allocate enough memory for struct fields ", fieldCount, " entries");
 
     for (uint32_t fieldIndex = 0; fieldIndex < fieldCount; ++fieldIndex) {
-        Type fieldType;
-        WASM_PARSER_FAIL_IF(!parseValueType(m_info, fieldType), "can't get ", fieldIndex, "th field Type");
+        StorageType fieldType;
+        WASM_PARSER_FAIL_IF(!parseStorageType(fieldType), "can't get ", fieldIndex, "th field Type");
+
+        // FIXME: Packed types in structs are temporarily unsupported
```

Comment:
It might be good to already open a bug for this and put the bug URL here in the next line of the comment (just mentioning this as I think upstream may request this anyway).